### PR TITLE
chore(lint) Added propTypes to all components

### LINKED ISSUE #21: Fix all lint-warnings
Mostly missing PropTypes. Should be fixed to avoid developer errors.

### PR DESCRIPTION
**Pull request by ivarconr**
screenshot from 2018-02-04 21-36-27

:fire: :fireworks: :fire: :fire_engine: 

closes #21

### CODE REVIEW

File: src/component/feature/form/strategy-input-percentage.jsx [hunk 1]
```diff
@@ -8,11 +9,19 @@ const labelStyle = {
...
     <div style={{ marginBottom: '20px' }}>
         <div style={labelStyle}>
             {name}: {value}%
         </div>
         <Slider min={0} max={100} defaultValue={value} value={value} onChange={onChange} label={name} />
     </div>
 );
+
+InputPercentage.propTypes = {
+    name: PropTypes.string.isRequired,
```
Comment (ivarconr) [inline src/component/feature/form/strategy-input-percentage.jsx]: fix: not actually required.

File: src/component/feature/form/strategy-input-percentage.jsx [hunk 2]
```diff
@@ -8,11 +9,19 @@ const labelStyle = {
     fontSize: '12px',
 };
 
-export default ({ name, value, onChange }) => (
+const InputPercentage = ({ name, value, onChange }) => (
     <div style={{ marginBottom: '20px' }}>
         <div style={labelStyle}>
             {name}: {value}%
         </div>
         <Slider min={0} max={100} defaultValue={value} value={value} onChange={onChange} label={name} />
```
Comment (ivarconr) [inline src/component/feature/form/strategy-input-percentage.jsx]: off topic: defaultValue and value should not be used at the same time. Input fields are either controller or uncontrolled, ref https://reactjs.org/docs/forms.html#controlled-components

File: src/component/common/index.js [hunk 3]
```diff
@@ -89,6 +110,9 @@ export const TogglesLinkList = ({ toggles }) => (
             ))}
     </List>
 );
+TogglesLinkList.propTypes = {
+    toggles: PropTypes.array,
```
Comment (ivarconr) [inline src/component/common/index.js]: required

File: src/component/common/index.js [hunk 4]
```diff
@@ -65,6 +78,10 @@ export const FormButtons = ({ submitText = 'Create', onCancel }) => (
         </Button>
     </div>
 );
+FormButtons.propTypes = {
+    submitText: PropTypes.string,
+    onCancel: PropTypes.func,
```
Comment (ivarconr) [inline src/component/common/index.js]: required

**Comment by SimenB** [comment]
:tada:

What about using flow?

**Comment by ivarconr** [comment]
Could be, I just wanted those lint warns gone. I'm all for flowing it all,
but that's a different task ;)

**Pull request merged**